src/telliot_feeds/reporters/tips/listener/tip_listener_filter.py

```diff
@@ -0,0 +1,69 @@
+import ast
+import sys
+from typing import Any
+from typing import Iterable
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from clamfig.base import Registry
+from eth_abi import decode_single
+from eth_utils.conversions import to_bytes
+from web3 import Web3 as w3
+
+from telliot_feeds.feeds import CATALOG_FEEDS
+from telliot_feeds.feeds import DataFeed
+from telliot_feeds.feeds import DATAFEED_BUILDER_MAPPING
+from telliot_feeds.feeds import LEGACY_DATAFEEDS
+from telliot_feeds.queries.query import OracleQuery
+from telliot_feeds.reporters.tips import CATALOG_QUERY_IDS
+from telliot_feeds.reporters.tips.listener.tip_listener import TipListener
+
+
+class TipListenerFilter(TipListener):
+    def decode_typ_name(self, qdata: bytes) -> str:
+        try:
+            qtype_name: str = decode_single("(string,bytes)", qdata)[0]
```

Comment:
Won't this decode into the type string and parameter value bytes? So, `qtype_name, _ = decode_single...`?